cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -160,19 +88,42 @@ result_t compute_kernel_dense_impl<Float>::operator()(const descriptor_t& desc,
     }
 
     if (desc.get_result_options().test(result_options::cov_matrix)) {
-        auto [cov, cov_event] =
-            compute_covariance(q_, rows_count_global, xtx, sums, bias, { gemm_event });
+        auto cov = pr::ndarray<Float, 2>::empty(q_, { column_count, column_count }, alloc::device);
+        sycl::event copy_event;
+        {
+            ONEDAL_PROFILER_TASK(copy_cov, q_);
+            copy_event = copy(q_, cov, xtx, { sums_event, gemm_event });
+        }
+
+        sycl::event cov_event;
+        {
+            ONEDAL_PROFILER_TASK(compute_covariance_matrix, q_);
+            cov_event = pr::covariance(q_, rows_count_global, sums, cov, bias, { copy_event });
+        }
+
```

Comment:
What about moving it to a completelyt separate function?

Reply:
In which folder? Just because looks like its bad to add a new function in primitives, because we usually return just events from pr:: functions. And for Cov and for Pca there is no single place to define previous functions liek compute_means

Reply:
Its related to all deleted wrappers

Reply:
as solution, we can add misc.hpp in pca/backend/gpu/ and cov/backend/gpu/ and move all duplications into these 2 files

Reply:
I think it would be the oprimal solution! Thanx